src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -265,26 +265,24 @@ trait Unapplies extends ast.TreeDSL {
    * ClassDef of the case class.
    */
   def caseClassCopyMeth(cdef: ClassDef): Option[DefDef] = {
-    def isDisallowed(vd: ValDef) = isRepeatedParamType(vd.tpt) || isByNameParamType(vd.tpt)
-    val classParamss  = constrParamss(cdef)
-
-    if (cdef.symbol.hasAbstractFlag || mexists(classParamss)(isDisallowed)) None
-    else {
+    val classParamss = constrParamss(cdef)
+    def warn() = runReporting.warning(cdef.namePos, "case `copy` method is allowed to have by-name parameters under Scala 3 (or with -Xsource-features:case-copy-by-name)", Scala3Migration, cdef.symbol)
+    def isDisallowed(vd: ValDef) = isRepeatedParamType(vd.tpt) || isByNameParamType(vd.tpt).tap(if (_) warn())
+    def copyOK = currentRun.sourceFeatures.caseCopyByName || !mexists(classParamss)(isDisallowed)
```

Comment:
now it tries to generate `copy` for repeated params:

```
➜ sandbox scpr 10770 T.scala -Vprint:cleanup -Xsource:3-cross
> scala compile --server=false -S 2.13.15-bin-914c657-SNAPSHOT -d . -release 8 --repository https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots T.scala -Vprint:cleanup -Xsource:3-cross
/Users/luc/code/scala/scala13/sandbox/T.scala:1: error: type mismatch;
 found   : Seq[Int]
 required: Int
case class C(x: Int)(y: Int*)
                     ^
1 error
```

Reply:
thanks, oops, I'll rename it to `sourceFeatures.anythingGoes`.